### PR TITLE
Handle recursive hashes

### LINKED ISSUE #608: How to use walletobject for save to android pay api?
I am following https://developers.google.com/save-to-android-pay/samples/quickstart-ruby guide which uses https://github.com/android-pay/s2ap-quickstart-ruby this repository has not been updated to use the new version of google-api-client. I tested it and It still works with the old version  client (<0.8).

I am trying to migrate this to the new version before implementing it in my system. However, I am not sure if the file https://github.com/android-pay/s2ap-quickstart-ruby/blob/master/wobs-discovery.json would work with the new file generation discovery. Currently when I run it I get a 'stack level too deep' error.

Any idea? I'm really not sure where to ask about this.

### PR DESCRIPTION
**Pull request by bantini**
Extension of pull request #609 with test case
Fixes #608

### CODE REVIEW

**Comment by bantini** [comment]
Signed cla with github email.

**Comment by bantini** [comment]
Signed cla.

**Comment by bantini** [comment]
I signed it!

File: spec/fixtures/files/test_api.json [hunk 1]
```diff
@@ -131,6 +134,49 @@
...
+      },
+      "string": {
+       "type": "string",
+       "description": "The string value."
+      }
+    },
+   "additionalProperties": {
+    "$ref": "RecursiveHat"
+   }
+  },
```
Comment (saicheems) [inline spec/fixtures/files/test_api.json]: Something is weird about the indenting here.

File: spec/fixtures/files/test_api.json [hunk 2]
```diff
@@ -131,6 +134,49 @@
         ]
       }
     },
+    "RecursiveHat": {
```
Comment (saicheems) [inline spec/fixtures/files/test_api.json]: Do tests fail for this schema without the change to `model.rb` and `_class.tmpl`?

Comment (bantini) [reply spec/fixtures/files/test_api.json]: Tests fail without the change to model.rb, but not for _class.tmpl.

File: spec/fixtures/files/test_api.json (hunk 1 above)
Comment (bantini) [reply spec/fixtures/files/test_api.json]: Fixed in the latest commit.

**Pull request merged**